File: backend/src/main/java/com/example/campuspetfoster/entity/Pet.java
package com.example.campuspetfoster.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

@Data
@TableName("pet")
public class Pet {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long ownerId;
    private String petName;
    private String species;
    private String breed;
    private String gender;
    private Integer age;
    private String photoUrl;
    private String healthNote;
    private String dietNote;
    private String vaccineStatus;
    private String dewormStatus;
}

File: frontend/src/api/pet.ts
import axios from './axios'

export interface Pet {
  id: number
  ownerId: number
  petName: string
  species: string
  breed: string
  gender: string
  age: number
  photoUrl: string
  healthNote: string
  dietNote: string
  vaccineStatus: string
  dewormStatus: string
}

export interface PetListParams {
  page?: number
  size?: number
  petName?: string
  species?: string
  vaccineStatus?: string
}

export interface PetListResponse {
  total: number
  list: Pet[]
}

export const getPetList = (params: PetListParams = {}): Promise<PetListResponse> => {
  return axios.get('/pets', { params })
}

export const getMyPets = (): Promise<Pet[]> => {
  return axios.get('/pets/my')
}

export const getPetById = (id: number): Promise<Pet> => {
  return axios.get(`/pets/${id}`)
}

export const createPet = (data: Partial<Pet>): Promise<Pet> => {
  return axios.post('/pets', data)
}

export const updatePet = (id: number, data: Partial<Pet>): Promise<Pet> => {
  return axios.put(`/pets/${id}`, data)
}

export const deletePet = (id: number): Promise<string> => {
  return axios.delete(`/pets/${id}`)
}

export const getSpeciesStats = (): Promise<Record<string, number>> => {
  return axios.get('/pets/stats/species')
}

export const getVaccineStats = (): Promise<Record<string, number>> => {
  return axios.get('/pets/stats/vaccine')
}
